src/llvm-gcroot.cpp

```diff
@@ -461,11 +463,59 @@ unsigned find_space_for(CallInst *callInst,
     return n;
 }
 
+void rearrangeRoots()
+{
+    for (auto BB = F.begin(), E(F.end()); BB != E; BB++) {
+        auto terminst = BB->getTerminator();
+        if (!dyn_cast<ReturnInst>(terminst) &&
+            !dyn_cast<UnreachableInst>(terminst))
```

Comment:
would be clearer to use `isa` here directly (the implementation of `dyn_cast` is `isa<T>(x) ? cast<T>(x) : NULL`)